play-media-native/screens/CreateMedia.tsx

```diff
@@ -1,91 +1,112 @@
+import { Screen } from '../features/Screen/Screen';
+import { getInitialStateFromOverrides } from '../helpers/contentItemHelper';
 import { getFileType } from '../helpers/media';
+import { generateID } from '../helpers/uuid';
+import { useContentItems } from '../hooks/useContentItems/useContentItems';
 import { useMediaQuery } from '../hooks/useMediaQuery/useMediaQuery';
 import { Media } from '../interfaces/media';
 import { styles } from '../theme/styles';
 import { theme } from '../theme/theme';
 
 export const CreateMediaScreen = ({ navigation, route }) => {
-  const [createdImage, setCreatedImage] = useState<Media>();
+  const [stateKey] = useState<string>(generateID());
+  const { contentItems, init, reset } = useContentItems();
+  const media = contentItems[stateKey] as Media;
+
+  const headerTitle = contentItems[stateKey]?.name || 'Untitled media';
 
+  const [createdImage, setCreatedImage] = useState<Media>();
+  const [mediaID, setMediaID] = useState(null);
+  const [mediaStatus, setMediaStatus] = useState(null);
   const [isValidating, setIsValidating] = useState(false);
-  const [showSuccessToast, setShowSuccessToast] = useState(false);
-  const [showErrorToast, setShowErrorToast] = useState(false);
-  const [shouldShowBottomActions, setShouldShowBottomActions] = useState(true);
-
-  const { refetch: refetchMediaListing } = useMediaQuery();
-
-  const onNameChange = useCallback((text: string) => {
-    setCreatedImage((prev) => ({
-      ...prev,
-      name: text,
-    }));
-  }, []);
-
-  const onDescriptionChange = useCallback((text: string) => {
-    setCreatedImage((prev) => ({
-      ...prev,
-      description: text,
-    }));
-  }, []);
+  const [isMediaItemSaved, setIsMediaItemSaved] = useState(false);
+  const [showSuccessView, setShowSuccessView] = useState(false);
+  const [showErrorView, setShowErrorView] = useState(false);
+
+  const { refetch: refetchMediaListing } = useMediaQuery(mediaID, mediaStatus);
 
   const handlePublish = useCallback(async () => {
     setIsValidating(true);
 
-    await uploadSingleImage({ ...createdImage, stateField: '', stateId: '', uploadStatus: '' })
+    await uploadSingleImage({
+      ...createdImage,
+      ...media,
+      stateField: '',
+      stateId: '',
+      uploadStatus: '',
+    })
       .then(async (uploadedImage) => {
         await publishMediaItem(uploadedImage.id)
           .then(async () => {
-            setShowSuccessToast(true);
+            setIsMediaItemSaved(true);
+            setMediaID(uploadedImage.id);
+            setMediaStatus(ITEM_STATUS.PUBLISHED);
             await refetchMediaListing();
             setIsValidating(false);
-            navigation.navigate('MainTabs');
+            setShowSuccessView(true);
+            setTimeout(() => {
+              navigation.navigate('MainTabs');
+            }, MEDIA_UPDATED_SUCCESSFULLY_TIMEOUT);
           })
           .catch(() => {
```

Comment:
Not really needed unless you want to display an error specific to publishing.
Otherwise the outer catch will do.